packages/flutter/lib/src/widgets/image.dart

```diff
@@ -104,13 +104,44 @@ ImageConfiguration createLocalImageConfiguration(BuildContext context, { Size? s
 /// See also:
 ///
 ///  * [ImageCache], which holds images that may be reused.
+///  * [precacheImageWithConfig], which this method delegates to.
 Future<void> precacheImage(
   ImageProvider provider,
   BuildContext context, {
   Size? size,
   ImageErrorListener? onError,
 }) {
-  final ImageConfiguration config = createLocalImageConfiguration(context, size: size);
+  return precacheImageWithConfig(
+    provider,
+    config: createLocalImageConfiguration(context, size: size),
+    onError: onError,
+  );
+}
+
+/// Prefetches an image into the image cache.
+///
+/// Returns a [Future] that will complete when the first image yielded by the
+/// [ImageProvider] is available or failed to load.
+///
+/// This method works identically to [precacheImage], except that it takes an
+/// [ImageConfiguration] directly instead of constructing one from a [BuildContext].
+/// By default, this value is [ImageConfiguration.empty].
+///
+/// In order to cache images which have asset variants, the correct device
+/// pixel ratio would need to be provided. For example, if there are 2x and
+/// 3x variants of an image, supplying an image configuration with a devicePixelRatio
+/// of 3.1 would select the larger variant. Leaving the default value of `1` would
+/// select the 2x variant.
```

Comment:
This sounds like it would turn this API into an easy footgun. To determine the correct dpr you likely need to look it up from the context anyways.

What problem are we trying to solve with this API?

Reply:
Many images don't have any variants, but good point. i can cache other assets differently, or just write a specialized caching method in my usecase.